File: src/06.hashMap/08.532.go
package hashmap

import (
	"math"
	"strconv"
)

func FindPairs(nums []int, k int) int {
	set := map[string]bool{}
	count := 0
	for i, v := range nums {
		for j := i + 1; j < len(nums); j++ {
			key1 := strconv.Itoa(v) + "-" + strconv.Itoa(nums[j])
			key2 := strconv.Itoa(nums[j]) + "-" + strconv.Itoa(v)
			if int(math.Abs(float64(v-nums[j]))) == k && (!set[key1] || !set[key2]) {
				count++
				set[key1] = true
				set[key2] = true
			}
		}
	}
	return count
}

File: src/06.hashMap/08.532.ts
export function findPairs(nums: number[], k: number): number {
    const set = new Set<string>();
    let count = 0;
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (
                Math.abs(nums[i] - nums[j]) === k &&
                (!set.has(`${nums[i]}-${nums[j]}`) ||
                    !set.has(`${nums[j]}-${nums[i]}`))
            ) {
                count++;
                set.add(`${nums[i]}-${nums[j]}`);
                set.add(`${nums[j]}-${nums[i]}`);
            }
        }
    }
    return count;
}
